sunpy/visualization/imageanimator.py

```diff
@@ -1001,25 +1001,60 @@ def _get_main_axes(self):
         self._set_unit_in_axis(axes)
         return axes
 
-    def _set_unit_in_axis(self, axes):
-        if self.unit_x_axis is not None:
-            axes.coords[2].set_format_unit(self.unit_x_axis)
-            axes.coords[2].set_ticks(exclude_overlapping=True)
-        if self.unit_y_axis is not None:
-            axes.coords[1].set_format_unit(self.unit_y_axis)
-            axes.coords[1].set_ticks(exclude_overlapping=True)
+    def _set_image_axes_units(self, unit_x_axis=None, unit_y_axis=None):
+        if(unit_x_axis is not None):
+            self.unit_x_axis = u.Unit(unit_x_axis)
+        else:
+            self.unit_x_axis = self.wcs.wcs.cunit[0]
+
+        if(unit_y_axis is not None):
+            self.unit_y_axis = u.Unit(unit_y_axis)
+        else:
+            self.unit_y_axis = self.wcs.wcs.cunit[1]
 
     def plot_start_image(self, ax):
         """Sets up plot of initial image."""
         imshow_args = {'interpolation': 'nearest',
                        'origin': 'lower',
                        }
         imshow_args.update(self.imshow_kwargs)
+        ax.ticklabel_format(style='sci')
         im = ax.imshow(self.data[self.frame_index], **imshow_args)
         if self.if_colorbar:
             self._add_colorbar(im)
         return im
 
+    def plot_user_input(self, axes, tick_spacing, x_tick_position = None, y_tick_position = None,  xlabel = None, ylabel = None ):
+        """"Lets the user set the tick spacing and tick labels with proper tick position""""
+        self._set_image_axis_units(unit_x_axis, unit_y_axis)
+        if self.unit_x_axis is not None:
+            axes.coords[2].set_format_unit(self.unit_x_axis)
+        if self.unit_y_axis is not None:
+            axes.coords[1].set_format_unit(self.unit_y_axis)
+
+        axes.coords[2].set_ticks(spacing = tick_spacing * self.wcs.wcs.cunit[0], exclude_overlapping=True)
+        axes.coords[1].set_ticks(spacing = tick_spacing * self.wcs.wcs.cunit[1], exclude_overlapping=True)
```

Comment:
I am a little bit not sure in this part, since if `self.wcs.wcs.cunit[1]` = u.degree , spacing can be u.degree, u.arcsec, u.arcmin.

So which to choose depends upon the user , but also user should not write in an invalid unit for tick_spacing

Reply:
Do we need a different unit for spacing as the one supplied by the user?  Can we not just use that for the spacing too?

If a user supplies an incompatible unit you can check for that and raise an error or just let the code crash.  Presumably there is somewhere where data is converted to the user's choice.  If the unit it incompatible, an ```InconsistentUnitError``` will probably be raised?

Reply:
We can use the user input, but the spacing will depend on the dimension of the input.

There are two cases 
* user input = None
   use some default spacing = wcs.cunit

* user input not None
   for example if user input u.degree, if we use that in spacing, that might cause a large spacing 
   between two adjacent labels, which the user might not want. He may want a smaller spacing as 
   supported by u.arcsec or u.arcmin

Reply:
Ah, I see.  Why not make the the tick spacing arg be a ```Quantity```.  Then the user automatically has to specify the unit they want?

Reply:
Yeah that will be a good way